rpc/server.go

```diff
@@ -39,178 +40,84 @@ func NewRpcServer(nitroClient *nitro.Client, chainId *big.Int, logger zerolog.Lo
-			rpcRequest.Params.Nonce,
-			rpcRequest.Params.AppDefinition,
-		)
-
-		rs.client.IncomingObjectiveRequests() <- objectiveRequestWithChan
-
-		objRes := rpcRequest.Params.Response(*rs.client.Address)
-		msg := serde.NewJsonRpcResponse(rpcRequest.Id, objRes)
-		messageData, err := json.Marshal(msg)
-		if err != nil {
-			panic("Could not marshal direct fund response message")
-		}
-
-		return messageData
+	err = subcribeToRequest(rs, serde.DirectDefundRequestMethod, func(obj directdefund.ObjectiveRequest) protocols.ObjectiveId {
+		return rs.client.CloseLedgerChannel(obj.ChannelId)
 	})
+
 	if err != nil {
-		panic(err)
+		return err
 	}
 
-	err = rs.connection.Subscribe(serde.VirtualDefundRequestMethod, func(data []byte) []byte {
-		rs.logger.Trace().Msgf("Rpc server received request: %+v", data)
-
-		rpcRequest := serde.JsonRpcRequest[virtualdefund.ObjectiveRequest]{}
-		err := json.Unmarshal(data, &rpcRequest)
-		if err != nil {
-			panic("could not unmarshal virtual defund objective request")
-		}
-
-		// todo: objective request is redefined so that it has a valid objectiveStarted channel.
-		// 	Should find a better way to accomplish this.
-		objectiveRequestWithChan := virtualdefund.NewObjectiveRequest(
-			rpcRequest.Params.ChannelId,
-		)
-
-		rs.client.IncomingObjectiveRequests() <- objectiveRequestWithChan
-
-		objRes := rpcRequest.Params.Id(*rs.client.Address, rs.chainId)
-		msg := serde.NewJsonRpcResponse(rpcRequest.Id, objRes)
-		messageData, err := json.Marshal(msg)
-		if err != nil {
-			panic("Could not marshal direct fund response message")
-		}
-
-		return messageData
+	err = subcribeToRequest(rs, serde.VirtualFundRequestMethod, func(obj virtualfund.ObjectiveRequest) virtualfund.ObjectiveResponse {
+		return rs.client.CreateVirtualPaymentChannel(obj.Intermediaries, obj.CounterParty, obj.ChallengeDuration, obj.Outcome)
 	})
+
 	if err != nil {
-		panic(err)
+		return err
 	}
 
-	err = rs.connection.Subscribe(serde.PayRequestMethod, func(data []byte) []byte {
-		rs.logger.Trace().Msgf("Rpc server received request: %+v", data)
-
-		rpcRequest := serde.JsonRpcRequest[serde.PaymentRequest]{}
-		err := json.Unmarshal(data, &rpcRequest)
-		if err != nil {
-			panic("could not unmarshal pay objective request")
-		}
-
-		rs.client.Pay(rpcRequest.Params.Channel, big.NewInt(int64(rpcRequest.Params.Amount)))
-
-		// TODO: What should we return here? A voucher?
-		msg := serde.NewJsonRpcResponse(rpcRequest.Id, rpcRequest.Params)
-		messageData, err := json.Marshal(msg)
-		if err != nil {
-			panic("Could not marshal pay response message")
-		}
-
-		return messageData
+	err = subcribeToRequest(rs, serde.VirtualDefundRequestMethod, func(obj virtualdefund.ObjectiveRequest) protocols.ObjectiveId {
+		return rs.client.CloseVirtualChannel(obj.ChannelId)
 	})
-	if err != nil {
-		panic(err)
-	}
-}
 
-// handleError "handles" an error by panicking
-func handleError(err error) {
 	if err != nil {
-		panic(err)
+		return err
 	}
```

Comment:
Curious why you decided to remove `handleError` after all?

Reply:
A couple reasons:
- A `handleError` function that panics is misleading.
- There is not a unifying error strategy for the rpc package. There are panics mixed with returned errors.

In general, the golang approach is to bubble up an error to the outermost layer. And I suggest that's what we should adopt for the rpc package.